pkg/client/client.go

```diff
@@ -317,6 +320,51 @@ func (c *synseClient) Read(rack, board, device string) (*scheme.Read, error) {
 	return out, nil
 }
 
+// ReadCached gets and parses the response from Synse Server's "readcached" endpoint.
+// Since the returned response is stream JSON, have to use a different way to
+// get these values out.
+// FIXME: Maybe there's a better way to do this.
+func (c *synseClient) ReadCached(params scheme.ReadCachedParams) ([]scheme.ReadCached, error) {
+	var readings []scheme.ReadCached
+	errScheme := new(scheme.Error)
+
+	client, err := newVersioned()
+	if err != nil {
+		return nil, err
+	}
+
+	// Check for Synse Server's failure response.
+	r, e := client.Get(readCachedBaseURI).Receive(nil, errScheme)
```

Comment:
its been a while since I looked at how the client handles things, but it looks like we are making two requests to `/readcached` here -- one to check the error state, the other to get the actual readings. I think we'd want this to be done in one request

if there is no error, the first request will get data returned. there can be a lot of data, so its pretty inefficient to request it all again. I think we can remove this error check request. Below, where we have the response from `httpClient.Do`, we should be able to check the HTTP code on the response -- if it is not OK, then we can marshal the JSON into the error struct, otherwise decode it into the `ReadCachced` struct

Reply:
Yes, that makes sense.

Also, seems like synse server client is used in two places at the moment: here and blackbox. As you mentioned in [blackbox's comments](https://github.com/vapor-ware/blackbox/blob/18ffe3927ed0dd81580695b4ddd9cb118148e8c9/test/e2e/api_test.go#L109), it's might be worthwhile to have a separate Go client that is usable in different places. Not sure if it's a high priority right now but something to think about.